(/social-connections) align with dashboard terminology; use partials (#1855)
clerk/clerk-docs: docs/authentication/social-connections/twitch.mdx

---
title: Add Twitch as a social connection
description: Learn how to allow users to sign up and sign in to your Clerk app with their Twitch account with OAuth.
---

<TutorialHero
  beforeYouStart={[
    {
      title: "A Clerk app is required.",
      link: "/docs/quickstarts/setup-clerk",
      icon: "clerk",
    },
    {
      title: "A Twitch account is required.",
      link: "https://www.twitch.tv/signup",
      icon: "user-circle",
    }
  ]}
>
  - Use Twitch to authenticate users with OAuth
</TutorialHero>

Enabling OAuth with [Twitch](https://dev.twitch.tv/docs/api/reference#oauth-client-credentials-flow) allows your users to sign up and sign in to your Clerk application with their Twitch account.

 You must have Two-Factor Authentication (2FA) enabled on your Twitch account to setup OAuth. To enable 2FA:

>  1. In the top-right, select your profile icon and select **Settings**.

Author: ```suggestion
 1. In the top-right of the [Twitch Developer console](https://dev.twitch.tv/console), select your avatar and select **Account Settings**.
```

>  1. In Settings, select the **Security & Privacy** tab.

Author: ```suggestion
 1. Select the **Security & Privacy** tab.
```

 1. Under **Security**, select **Set Up Two-Factor Authentication**.
 1. Select **Enable 2FA**. A modal will open.
 1. Enter your phone number and select **Continue**.
 1. Enter the code sent to your phone and select **Continue**.
 1. If you want to use an authenticator app, follow the instructions on the screen. Otherwise, select **Skip & Use SMS**.
 1. Select **Done**.

## Configure for your development instance

For _development instances_, Clerk uses preconfigured shared OAuth credentials and redirect URIs—no other configuration is needed.

1. In the Clerk Dashboard, navigate to the [**SSO connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
1. Select **Add connection** and select **For all users**.
1. In the **Choose provider** dropdown, select **Twitch**.
1. Select **Add connection**.

## Configure for your production instance

For _production instances_, you must provide custom credentials.

> To make the setup process easier, it's recommended to keep two browser tabs open: one for the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [Twitch Developers](https://dev.twitch.tv/console) page.

Author: ```suggestion
To make the setup process easier, it's recommended to keep two browser tabs open: one for the [Clerk Dashboard](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) and one for your [Twitch Developer Console](https://dev.twitch.tv/console).
```

<Steps>
  ### Enable Twitch as a social connection in Clerk

  1. In the Clerk Dashboard, navigate to the [**SSO connections**](https://dashboard.clerk.com/last-active?path=user-authentication/sso-connections) page.
  1. Select **Add connection** and select **For all users**.
  1. In the **Choose provider** dropdown, select **Twitch**.
  1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on.
  1. Save the **OAuth Redirect URL** somewhere secure. Keep the modal and page open.

    ### Create a Twitch Developer app

>   1. On a separate page, go to the [Twitch Developers](https://dev.twitch.tv/console) page and sign in.
>   1. In the sidebar, select **Applications**.

Author: ```suggestion
  1. In the left sidebar of the [Twitch Developer Console](https://dev.twitch.tv/console), select **Applications**.
```

>   1. Select **Create an Application**. You'll be redirected to the [**Developer Applications**](https://dev.twitch.tv/console/apps) page.

Author: ```suggestion
  1. Select **Register Your Application**. You'll be redirected to the [**Register Your Application**](https://dev.twitch.tv/console/apps/create) page.
```

>   1. Select **Register Your Application**.

Author: ```suggestion
```

>   1. Complete the required fields. For **OAuth Redirect URLs**, paste the **OAuth Redirect URL** you saved from the Clerk Dashboard.

Author: ```suggestion
  1. Complete the required fields. Under **OAuth Redirect URLs**, paste the **OAuth Redirect URL** you saved from the Clerk Dashboard.
```

>   1. Select **Create**. You'll be redirected to the **Developer Applications** page where you can see your new application listed.
> 
>   ### Set the Client ID and Client Secret in the Clerk Dashboard
> 
>   1. In the **Developer Applications** page, select **Manage** next to the application you created. You'll be redirected to your app's settings page.
>   1. Select **New Secret**. Save this **Client Secret** and the **Client ID** somewhere secure.
>   1. Navigate back to the Clerk Dashboard where the modal should still be open. Paste the **Client ID** and **Client Secret** values that you saved into the respective fields.
>   1. Select **Add connection**.

Author: ```suggestion
  1. Select **Create**. You'll be redirected to the **Developer Applications** page where you can see your new application listed.
  1. Select **Manage** next to the application you created. You'll be redirected to your app's settings page.
  1. Select **New Secret**. Save the **Client Secret** and **Client ID** somewhere secure.

  ### Set the Client ID and Client Secret in the Clerk Dashboard
 
  <Include src="_partials/authentication/set-client-id-secret" />
```
